### PR TITLE
Heal breaks shield fix

### PR DESCRIPTION
**Pull request by wmann95**
## About the PR 
Fixes the problem of applying ointment to a damaged Urist wielding a shield utterly demolishing the shield. This problem was also causing the demolition of the shield when Urist was downed, gasping, and also wielding the shield. Both of these are fixed.

Also, fixes the problem of the stripping menu not being updated when the shield breaks.

**Changelog**
- fix: Wonderous buckler destruction upon heal, as well as destruction of buckler upon taking asphyxiation damage whilst down and wielding shield.
- fix: Broken shields not updating strip menu upon item destruction.

Closes: #10405
Closes: #10423

### CODE REVIEW

File: Content.Shared/Blocking/BlockingUserSystem.cs [hunk 1]
```diff
@@ -54,12 +54,29 @@ private void OnAnchorChanged(EntityUid uid, BlockingUserComponent component, ref
...
+    private bool IsDamageValid(DamageChangedEvent args)
+    {
+        if (args.DamageDelta == null)
+        {
+            return false;
+        }
+
+        return !(args.DamageDelta.DamageDict.ContainsKey("Asphyxiation") || args.DamageDelta.DamageDict.ContainsKey("Bloodloss") || args.DamageDelta.DamageDict.ContainsKey("Cellular")
+            || args.DamageDelta.DamageDict.ContainsKey("Poison") || args.DamageDelta.DamageDict.ContainsKey("Radiation"));
+    }
```
Comment (metalgearsloth) [inline Content.Shared/Blocking/BlockingUserSystem.cs:78]: This looks very hacky when it should be using damage resistances that already exist.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Should be fixing whatever is causing the riot shield to be destroyed upon negative damage being applied.

File: Content.Shared/Blocking/BlockingUserSystem.cs (hunk 1 above)
Comment (wmann95) [reply Content.Shared/Blocking/BlockingUserSystem.cs]: Agreed, I've removed this in the new commit, only keeping the check that the damage is positive, so no heals.

**Review by wmann95 — COMMENTED** [review]
I agreed with the reviewal and have removed the offending code.

File: Content.Shared/Blocking/BlockingUserSystem.cs [hunk 2]
```diff
@@ -54,7 +54,7 @@ private void OnAnchorChanged(EntityUid uid, BlockingUserComponent component, ref
 
     private void OnDamageChanged(EntityUid uid, BlockingUserComponent component, DamageChangedEvent args)
     {
-        if (component.BlockingItem != null)
+        if (component.BlockingItem != null && args.DamageIncreased)
```
Comment (mirrorcult) [inline Content.Shared/Blocking/BlockingUserSystem.cs:57]: shouldnt this be getting checked on the subscription for the blocking item? that way if you somehow directly heal the shield, it won't also break

File: Content.Shared/Blocking/BlockingSystem.cs [hunk 3]
```diff
@@ -40,10 +42,32 @@ public override void Initialize()
...
+            foreach(var hand in hands.Hands)
+            {
+                if (hand.Value.HeldEntity == uid)
+                {
+                    _handsSystem.TryDrop((EntityUid) component.User, hand.Value, null, true, true, hands);
+                    return;
+                }
+            }
+        }
+    }
```
Comment (mirrorcult) [inline Content.Shared/Blocking/BlockingSystem.cs:69]: this is definitely not the correct solution and won't work for anything else getting destroyed. it probably shouldn't be getting fixed in this PR regardless (as it seems to be a general bug with the stripping menu)

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Comment by mirrorcult** [comment]
changelog is also not formatted correctly, needs a :cl:
check other PRs or the PR template to see how its done

**Pull request closed without merging**